### PR TITLE
Unify fixed point & Interest Accrual

### PR DESCRIPTION
**Pull request by branan**

### CODE REVIEW

File: pallets/loans/src/math.rs
Comment (branan) [reply pallets/loans/src/math.rs:34]: This gets used in the tests a ton. Should probably get moved into the testutils module, but for now I've left it.

File: pallets/loans/src/functions.rs
Comment (offerijns) [inline pallets/loans/src/functions.rs:368]: I think this TODO can be removed
```suggestion
```

**Review by offerijns — APPROVED** [review]

**Comment by branan** [comment]
@NunoAlexandre This is the PR I pinged you about last week. There are two big changes:

1) Previously, the loans pallet used a custom FixedPoint number for holding currency amounts. This sort-of-simplified some intermediate computations but required a bunch of conversation to and from the `Balance` type. We now keep currency amounts in the `Balance` type just like every other pallet.

2) Instead of storing accumulated interest rates in each loan individually, the storage is now centralized. The goal is to get to a place where we can accumulate the rates in a separate operation from using them - that's not quite done yet and our benchmarks can't yet reflect the optimization, but it's a step in the right direction. Importantly, the storage for the loans pallet has been updated for it, so we won't need to make a big migration when we finish the work.

File: pallets/loans/src/math.rs
Comment (NunoAlexandre) [inline pallets/loans/src/math.rs:44]: I would leave a comment here explaining that it's meant to represent how much some other value should be incremented or decremented by. When I first read this, I thought this was meant to be Like `Inc(x) == x+1`.

File: pallets/loans/src/math.rs
Comment (NunoAlexandre) [inline pallets/loans/src/math.rs:306]:
```suggestion
		let debt: Balance = 100 * USD;
```

**Review by NunoAlexandre — APPROVED** [review]
Looks good to me, left minor comments but leave it up to you 👍

File: pallets/loans/src/math.rs
Comment (NunoAlexandre) [reply pallets/loans/src/math.rs:60]: That's a good point, it definitely reads better.

Comment (offerijns) [reply pallets/loans/src/math.rs:60]: Is it fine if we create a ticket for making this change in the future though @NunoAlexandre ? I don't think this should block merging this PR.

Comment (NunoAlexandre) [reply pallets/loans/src/math.rs:60]: Yeah, absolutely not a blocker, that sounds good to me 👍

File: pallets/loans/src/math.rs
Comment (offerijns) [reply pallets/loans/src/math.rs:45]: Done!

**Review by offerijns — DISMISSED** [review]

**Review by NunoAlexandre — APPROVED** [review]
💯 nice work!

**Review by NunoAlexandre — APPROVED** [review]

**Review by NunoAlexandre — DISMISSED** [review]

**Review by NunoAlexandre — APPROVED** [review]

**Comment by branan** [comment]
Changes since last review:
* Resolve merge conflict with RPC-related changes
* re-sign all commits since some early ones were done before we required signing

This just needs a new :+1: from somebody and it is good to go

**Pull request merged**
